components/sup/src/main.rs

```diff
@@ -73,24 +73,17 @@ static RING_ENVVAR: &'static str = "HAB_RING";
 static RING_KEY_ENVVAR: &'static str = "HAB_RING_KEY";
 
 fn main() {
-    if let Err(err) = start() {
-        println!("{}", err);
-        match err {
-            SupError {
-                err: Error::ProcessLocked(_),
-                ..
-            }
-            | SupError {
-                err: Error::Departed,
-                ..
-            }
-            | SupError {
-                err: Error::ButterflyError(_),
-                ..
-            } => process::exit(ERR_NO_RETRY_EXCODE),
-            _ => process::exit(1),
+    let result = start();
+    let exit_code;
+    match result {
+        Err(ref err) => {
+            println!("{}", err);
+            exit_code = ERR_NO_RETRY_EXCODE;
         }
-    }
+        Ok(_) => exit_code = 0,
+    };
```

Comment:
This whole bit can be written more simply as 
``` rust
let exit_code = match result {
    Ok(_) => 0,
    Err(ref err) => {
        ...
        ERR_NO_RETRY_EXCODE
    }
};
```

Reply:
Ooh, good call!